### ISSUE TITLE
Having two sorted associations in the same class generates two sort methods, which fails to compile

### ISSUE THREAD

**Issue by Nava2** [umple/umple#261]
Originally reported on Google Code with ID 362
Owned by `redeyedmars`

---

**What steps will reproduce the problem?**

// This example demonstrates sorted associations
// The main program adds items out of order
// But when they are printed the output will be sorted
//

class Academy {
  1 -- \* Course sorted {code};
  1 -- \* Student registrants sorted {id};
  public static void main(String [ ] args) {
    Academy ac = new Academy();
    ac.addRegistrant(12,"Jim");
    ac.addRegistrant(4,"Ali");
    ac.addRegistrant(8,"Mary");
    ac.addRegistrant(3,"Francois");
    ac.addCourse("CS191");
    ac.addCourse("SE200");
    ac.addCourse("EN100");
    System.out.println(ac);
  }
  public String toString() {
    String result="";
    for (Student s: getRegistrants()) {
      result+=s+"\n";
    }
    return result;
  }
}

class Student {
  Integer id;
  name;
  public String toString() {return("Student="+id+"["+name+"]");}
}

class Course {
  code;
  public String toString() {return("Course="+code);}
}

class Registration {
  // Derived delegated attribute used for sorting
  Integer id = {getStudent().getId()}

  // Derived delegated attribute used for sorting
  code = {getCourse().getCode()}
- sorted {id}-- 1 Student;
- sorted {code} -- 1 Course;
  }

**What is the expected output? What do you see instead?**

The following error is generated when compiling the Java:
"m.ump":237: error: method sort(List,String) is already defined in class Academy
  public void sort(List toSort, final String thePriority)
              ^

It should be fairly easy to ensure that only one sort method is output per class, with
the above as a test case. Check the same test case in PHP, Ruby etc.

---

Reported by @umple on 2013-01-03 15:41:15

**Comment by Nava2** [comment]
By the way, it should be possible to have two sorted associations in a class, so don't
just raise an Umple error here.

---

Reported by @umple on 2013-01-03 15:42:00

**Comment by Nava2** [comment]
Assigned to Geoffrey Guest since he has started work and reports he has made significant
progress.

---

Reported by @umple on 2013-01-06 13:47:15
- **Owner changed**: `redeyedmars`

**Comment by Nava2** [comment]
I have attached a zip folder with a jet file that fixes this problem. It goes in the
UmpleToTemplate project's templates. Also included are 4 test cases I have tried(with
varying number of associations 0 to 3).

I tried to create a patch using:
svn diff -x --ignore-eol-style > ~/patchGUEST-362-added_boolean_to_ensure_one_sort_method.diff

hope it worked, there were no messages.

---

Reported by `redeyedmars` on 2013-01-07 02:50:11
- _Attachment: [issue_362.zip](https://storage.googleapis.com/google-code-attachments/umple/issue-362/comment-3/issue_362.zip)_
